src/kernel-install/kernel-install.c

```diff
@@ -1019,25 +1020,52 @@ static bool bypass(void) {
 
 static int verb_add(int argc, char *argv[], void *userdata) {
         Context *c = ASSERT_PTR(userdata);
+        _cleanup_free_ char *vmlinuz = NULL;
+        const char *version, *kernel;
+        char **initrds;
+        struct utsname un;
         int r;
 
-        assert(argc >= 3);
         assert(argv);
 
         if (bypass())
                 return 0;
 
         c->action = ACTION_ADD;
 
-        r = context_set_version(c, argv[1]);
+        version = argc > 1 ? argv[1] : NULL;
+        kernel = argc > 2 ? argv[2] : NULL;
+        initrds = strv_skip(argv, 3);
+
+        if (!version) {
+                assert_se(uname(&un) >= 0);
+                version = un.release;
+        }
+
+        if (!kernel) {
+                vmlinuz = path_join("/usr/lib/modules/", version, "/vmlinuz");
+                if (!vmlinuz)
+                        return log_oom();
+
+                if (access(vmlinuz, F_OK) < 0) {
```

Comment:
Note, `context_set_kernel()` calls chase(), so the check here can be dropped.

Reply:
hmm, the reason i added this is to give clearer messages on what's goind on, but I think you are right we shouldn't duplicate the check fully: the fact that this follows symlinks here is kinda pointless, given we do that later anyway.

I will thus downgrade this to laccess(), i.e. not follow symlinks. that way, when auto-determining the kernel we'll just check if the inode is there, regardless if file or symlink. And then in the later logic we'll actually follow up with what it is.